rocprim/include/rocprim/thread/thread_search.hpp

```diff
@@ -61,8 +61,9 @@ ROCPRIM_HOST_DEVICE inline void merge_path_search(OffsetT        diagonal,
                                                   BinaryFunction compare_function
                                                   = BinaryFunction())
 {
-    path_coordinate.x = rocprim::detail::merge_path(a, b, a_len, b_len, diagonal, compare_function);
-    path_coordinate.y = diagonal - path_coordinate.x;
+    OffsetT split_min = rocprim::detail::merge_path(a, b, a_len, b_len, diagonal, compare_function);
+    path_coordinate.x = ::rocprim::min(split_min, a_len);
+    path_coordinate.y = diagonal - split_min;
```

Comment:
Should x and y be constrained to the ranges of [a, a_len] and [b, b_len] ?  Should that constraint be checked in merge_path(...) before returning?